clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -972,16 +975,28 @@ private Record nextFetchedRecord() {
         private List<ConsumerRecord<K, V>> fetchRecords(int maxRecords) {
             if (isFetched)
                 return Collections.emptyList();
+            if (nextInlineException != null) {
```

Comment:
As I said, the behavior was unintentionally lost. We can fix it here, or I can submit a separate patch before merging this to fix it. I am not debating the current behavior, merely saying that we should preserve it until we have an alternative has been discussed and agreed upon in the community.